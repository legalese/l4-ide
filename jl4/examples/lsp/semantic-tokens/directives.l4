ยง `Directives`

#EVAL 3 < 2

#EVAL 3 BELOW 2

#EVAL "bar" < "car"

#EVAL TRUE < FALSE

#CHECK TRUE

#CHECK 2 + 3

#EVAL foo WHERE foo MEANS 5
#CHECK foo WHERE foo MEANS 5

#EVAL foo 4 WHERE foo x MEANS x PLUS x
#CHECK foo "String" WHERE foo x MEANS x
