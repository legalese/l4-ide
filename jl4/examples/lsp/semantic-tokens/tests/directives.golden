Phase: Lexer
1:1-2 SemanticTokenTypes_Operator
1:3-15 SemanticTokenTypes_Variable
3:1-6 SemanticTokenTypes_Macro
3:7-8 SemanticTokenTypes_Number
3:9-10 SemanticTokenTypes_Operator
3:11-12 SemanticTokenTypes_Number
5:1-6 SemanticTokenTypes_Macro
5:7-8 SemanticTokenTypes_Number
5:9-14 SemanticTokenTypes_Keyword
5:15-16 SemanticTokenTypes_Number
7:1-6 SemanticTokenTypes_Macro
7:7-12 SemanticTokenTypes_String
7:13-14 SemanticTokenTypes_Operator
7:15-20 SemanticTokenTypes_String
9:1-6 SemanticTokenTypes_Macro
9:7-11 SemanticTokenTypes_Variable
9:12-13 SemanticTokenTypes_Operator
9:14-19 SemanticTokenTypes_Variable
11:1-7 SemanticTokenTypes_Macro
11:8-12 SemanticTokenTypes_Variable
13:1-7 SemanticTokenTypes_Macro
13:8-9 SemanticTokenTypes_Number
13:10-11 SemanticTokenTypes_Operator
13:12-13 SemanticTokenTypes_Number
15:1-6 SemanticTokenTypes_Macro
15:7-10 SemanticTokenTypes_Variable
15:11-16 SemanticTokenTypes_Keyword
15:17-20 SemanticTokenTypes_Variable
15:21-26 SemanticTokenTypes_Keyword
15:27-28 SemanticTokenTypes_Number
16:1-7 SemanticTokenTypes_Macro
16:8-11 SemanticTokenTypes_Variable
16:12-17 SemanticTokenTypes_Keyword
16:18-21 SemanticTokenTypes_Variable
16:22-27 SemanticTokenTypes_Keyword
16:28-29 SemanticTokenTypes_Number
18:1-6 SemanticTokenTypes_Macro
18:7-10 SemanticTokenTypes_Variable
18:11-12 SemanticTokenTypes_Number
18:13-18 SemanticTokenTypes_Keyword
18:19-22 SemanticTokenTypes_Variable
18:23-24 SemanticTokenTypes_Variable
18:25-30 SemanticTokenTypes_Keyword
18:31-32 SemanticTokenTypes_Variable
18:33-37 SemanticTokenTypes_Keyword
18:38-39 SemanticTokenTypes_Variable
19:1-7 SemanticTokenTypes_Macro
19:8-11 SemanticTokenTypes_Variable
19:12-20 SemanticTokenTypes_String
19:21-26 SemanticTokenTypes_Keyword
19:27-30 SemanticTokenTypes_Variable
19:31-32 SemanticTokenTypes_Variable
19:33-38 SemanticTokenTypes_Keyword
19:39-40 SemanticTokenTypes_Variable

Phase: Parse
1:1-2 SemanticTokenTypes_Operator
1:3-15 SemanticTokenTypes_Interface
3:1-6 SemanticTokenTypes_Macro
3:7-8 SemanticTokenTypes_Number
3:9-10 SemanticTokenTypes_Operator
3:11-12 SemanticTokenTypes_Number
5:1-6 SemanticTokenTypes_Macro
5:7-8 SemanticTokenTypes_Number
5:9-14 SemanticTokenTypes_Keyword
5:15-16 SemanticTokenTypes_Number
7:1-6 SemanticTokenTypes_Macro
7:7-12 SemanticTokenTypes_String
7:13-14 SemanticTokenTypes_Operator
7:15-20 SemanticTokenTypes_String
9:1-6 SemanticTokenTypes_Macro
9:7-11 SemanticTokenTypes_Variable
9:12-13 SemanticTokenTypes_Operator
9:14-19 SemanticTokenTypes_Variable
11:1-7 SemanticTokenTypes_Macro
11:8-12 SemanticTokenTypes_Variable
13:1-7 SemanticTokenTypes_Macro
13:8-9 SemanticTokenTypes_Number
13:10-11 SemanticTokenTypes_Operator
13:12-13 SemanticTokenTypes_Number
15:1-6 SemanticTokenTypes_Macro
15:7-10 SemanticTokenTypes_Variable
15:11-16 SemanticTokenTypes_Keyword
15:17-20 SemanticTokenTypes_Variable
15:21-26 SemanticTokenTypes_Keyword
15:27-28 SemanticTokenTypes_Number
16:1-7 SemanticTokenTypes_Macro
16:8-11 SemanticTokenTypes_Variable
16:12-17 SemanticTokenTypes_Keyword
16:18-21 SemanticTokenTypes_Variable
16:22-27 SemanticTokenTypes_Keyword
16:28-29 SemanticTokenTypes_Number
18:1-6 SemanticTokenTypes_Macro
18:7-10 SemanticTokenTypes_Variable
18:11-12 SemanticTokenTypes_Number
18:13-18 SemanticTokenTypes_Keyword
18:19-22 SemanticTokenTypes_Variable
18:23-24 SemanticTokenTypes_Variable
18:25-30 SemanticTokenTypes_Keyword
18:31-32 SemanticTokenTypes_Variable
18:33-37 SemanticTokenTypes_Keyword
18:38-39 SemanticTokenTypes_Variable
19:1-7 SemanticTokenTypes_Macro
19:8-11 SemanticTokenTypes_Variable
19:12-20 SemanticTokenTypes_String
19:21-26 SemanticTokenTypes_Keyword
19:27-30 SemanticTokenTypes_Variable
19:31-32 SemanticTokenTypes_Variable
19:33-38 SemanticTokenTypes_Keyword
19:39-40 SemanticTokenTypes_Variable

Phase: Check
1:1-2 SemanticTokenTypes_Operator
1:3-15 SemanticTokenTypes_Interface
3:1-6 SemanticTokenTypes_Macro
3:7-8 SemanticTokenTypes_Number
3:9-10 SemanticTokenTypes_Operator
3:11-12 SemanticTokenTypes_Number
5:1-6 SemanticTokenTypes_Macro
5:7-8 SemanticTokenTypes_Number
5:9-14 SemanticTokenTypes_Keyword
5:15-16 SemanticTokenTypes_Number
7:1-6 SemanticTokenTypes_Macro
7:7-12 SemanticTokenTypes_String
7:13-14 SemanticTokenTypes_Operator
7:15-20 SemanticTokenTypes_String
9:1-6 SemanticTokenTypes_Macro
9:7-11 SemanticTokenTypes_Class
9:12-13 SemanticTokenTypes_Operator
9:14-19 SemanticTokenTypes_Class
11:1-7 SemanticTokenTypes_Macro
11:8-12 SemanticTokenTypes_Class
13:1-7 SemanticTokenTypes_Macro
13:8-9 SemanticTokenTypes_Number
13:10-11 SemanticTokenTypes_Operator
13:12-13 SemanticTokenTypes_Number
15:1-6 SemanticTokenTypes_Macro
15:7-10 SemanticTokenTypes_Variable
15:11-16 SemanticTokenTypes_Keyword
15:17-20 SemanticTokenTypes_Variable
15:21-26 SemanticTokenTypes_Keyword
15:27-28 SemanticTokenTypes_Number
16:1-7 SemanticTokenTypes_Macro
16:8-11 SemanticTokenTypes_Variable
16:12-17 SemanticTokenTypes_Keyword
16:18-21 SemanticTokenTypes_Variable
16:22-27 SemanticTokenTypes_Keyword
16:28-29 SemanticTokenTypes_Number
18:1-6 SemanticTokenTypes_Macro
18:7-10 SemanticTokenTypes_Function
18:11-12 SemanticTokenTypes_Number
18:13-18 SemanticTokenTypes_Keyword
18:19-22 SemanticTokenTypes_Function
18:23-24 SemanticTokenTypes_Variable
18:25-30 SemanticTokenTypes_Keyword
18:31-32 SemanticTokenTypes_Variable
18:33-37 SemanticTokenTypes_Keyword
18:38-39 SemanticTokenTypes_Variable
19:1-7 SemanticTokenTypes_Macro
19:8-11 SemanticTokenTypes_Function
19:12-20 SemanticTokenTypes_String
19:21-26 SemanticTokenTypes_Keyword
19:27-30 SemanticTokenTypes_Function
19:31-32 SemanticTokenTypes_Variable
19:33-38 SemanticTokenTypes_Keyword
19:39-40 SemanticTokenTypes_Variable

